lib/ruby_lsp/requests/folding_ranges.rb

```diff
@@ -195,7 +195,8 @@ def add_call_range(node)
             receiver = receiver.receiver
           when SyntaxTree::MethodAddBlock
             visit(receiver.block)
-            receiver = receiver.call.receiver
+            receiver = receiver.call
+            receiver = receiver.receiver unless receiver.is_a?(SyntaxTree::FCall)
```

Comment:
Will `if receiver.respond_to?(:receiver)` be more flexible?

Reply:
I don't think it will type check properly tho

Reply:
I wonder if we do the reverse and list explicitly the cases where we're going to look for `receiver`?

Something like:

```rb
case receiver
when CallKind1, CallKind2
  receiver.receiver
end
```

To be more robust if `syntax-tree` add new node kinds.

Reply:
I like the idea of an allow list approach. Using this made me realize that `Command` is also a possible node, but doesn't implement `receiver` because it's invoked on `self`.

I have started to check for the nodes we care about specifically.